### PR TITLE
Add developer document for "Version Support Policy"

### LINKED ISSUE #1775: Updates not prevented when WooCommerce is outdated
In PR #1747 it was identified that bumping the `WC requires at least` value may have dangerous consequences for a site updating WC Pay but remaining on an old version of WooCommerce.

A potential solution was identified in https://github.com/Automattic/woocommerce-payments/pull/1747#issuecomment-836709557 

> The solution could be trying to hook into something like upgrader_pre_download or some other upgrader hook, fetch the updated plugin information, and stop the download/install before it happens. Interestingly, there aren't many resources on the internet that would describe how to solve such a scenario. If we were to work on this, the following test cases come to mind:

> auto update
> manual update
> manual bulk update with other plugins
> update on multisite

### PR DESCRIPTION
**Pull request by htdat**
Fixes #1775

#### Changes proposed in this Pull Request

- Add developer document for "Version Support Policy". 
- Also add the introduction in the main README.md.

#### Testing instructions

* Review the content of the new doc and ensure they're correct and helpful for contributors (developers).
* Recommend using this rendered link to review https://github.com/Automattic/woocommerce-payments/blob/add/version-support-policy/docs/version-support-policy.md

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 

**Post merge**

- [x] ~Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_~

### CODE REVIEW

**Review by jrodger — APPROVED** [review]
Looks great @htdat, thanks for pulling together all the other various pieces of documentation here.

File: docs/version-support-policy.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+For WooCommerce, we are still catching up to fully support this L-2 policy. The real minimum WooCommerce version can be found on [`woocommerce-payments.php`](https://github.com/Automattic/woocommerce-payments/blob/develop/woocommerce-payments.php) with the `WC requires at least` tag. There are two main reasons for this catch-up:
+
+- There is no reliable built-in functionality to prevent a merchant to install the latest version of WooCommerce Payments (WCPay) with an unsupported WooCommerce version. To deal with this issue, we continue loading WCPay if sites have an active WCPay account, but nudge their owners to upgrade WooCommerce. More details in [this PR](https://github.com/Automattic/woocommerce-payments/pull/3010), which is later refactored to `WC_Payments_Dependency_Service`.
+- Some of our internal consumers are also not catching up with using the latest version of WooCommerce.
```
Comment (jessy-p) [inline docs/version-support-policy.md:10]: Some of our dependent plugins (?) - if that is what is intended here

Comment (jessy-p) [inline docs/version-support-policy.md:10]: are also not yet using

File: docs/version-support-policy.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+However, in the practice, we're really enforcing only WordPress core with the `Requires at least` in [`readme.txt`](https://github.com/Automattic/woocommerce-payments/blob/develop/readme.txt) and [`woocommerce-payments.php`](https://github.com/Automattic/woocommerce-payments/blob/develop/woocommerce-payments.php).
+
+For WooCommerce, we are still catching up to fully support this L-2 policy. The real minimum WooCommerce version can be found on [`woocommerce-payments.php`](https://github.com/Automattic/woocommerce-payments/blob/develop/woocommerce-payments.php) with the `WC requires at least` tag. There are two main reasons for this catch-up:
+
+- There is no reliable built-in functionality to prevent a merchant to install the latest version of WooCommerce Payments (WCPay) with an unsupported WooCommerce version. To deal with this issue, we continue loading WCPay if sites have an active WCPay account, but nudge their owners to upgrade WooCommerce. More details in [this PR](https://github.com/Automattic/woocommerce-payments/pull/3010), which is later refactored to `WC_Payments_Dependency_Service`.
```
[the remark this answers is not in the document]
Comment (htdat) [reply docs/version-support-policy.md]: Updated: ab4e263abf39ef47483992361d9fc7445442175b

File: docs/version-support-policy.md (hunk 1 above)
Comment (htdat) [reply docs/version-support-policy.md]: Updated here with a private info: ab4e263abf39ef47483992361d9fc7445442175b

Comment (htdat) [reply docs/version-support-policy.md]: Updated here: ab4e263abf39ef47483992361d9fc7445442175b

**Review by jessy-p — APPROVED** [review]

**Comment by htdat** [comment]
Thanks for your review @jrodger and @jessy-p. 

@jessy-p - I've amended some changes based on your feedback. Can you have another look before I merge this PR?

**Pull request merged**